gateway/engine/core/src/main/java/io/apiman/gateway/engine/impl/EngineImpl.java

```diff
@@ -64,6 +67,7 @@ public EngineImpl(final IRegistry registry, final IPluginRegistry pluginRegistry
         
         policyFactory.setPluginRegistry(pluginRegistry);
         metrics.setComponentRegistry(componentRegistry);
+        initializeMetric(metrics);
```

Comment:
Can you call initialize on all of the engine's dependencies here instead of just the metrics one?  It'd be nice to have that option for everything.